src/libcore/slice.rs

```diff
@@ -914,6 +951,27 @@ impl<'a, T> IterMut<'a, T> {
     pub fn into_slice(self) -> &'a mut [T] {
         make_mut_slice!(T => &'a mut [T]: self.ptr, self.end)
     }
+
+    // Helper function for IterMut::nth
+    fn iter_nth(&mut self, n: usize) -> Option<&'a mut T> {
+        match make_mut_slice!(T => &'a mut [T]: self.ptr, self.end).get_mut(n) {
+            Some(elem_ref) => if mem::size_of::<T>() == 0 {
```

Comment:
The logic here (especially with 0-sized types) is unfortunately duplicated with the `next` functions, would it be possible to deduplicate these conditionals?